sys-apps/grep/grep-3.6.recipe

```diff
@@ -5,50 +5,61 @@ matching lines."
 HOMEPAGE="https://www.gnu.org/software/grep/"
 COPYRIGHT="1992-2016 Free Software Foundation, Inc."
 LICENSE="GNU GPL v3"
-REVISION="1"
+REVISION="2"
 SOURCE_URI="http://ftp.gnu.org/gnu/grep/grep-$portVersion.tar.xz"
 CHECKSUM_SHA256="667e15e8afe189e93f9f21a7cd3a7b3f776202f417330b248c2ad4f997d9373e"
 PATCHES="grep-$portVersion.patchset"
 
-ARCHITECTURES="?x86_gcc2 x86 x86_64 ?arm"
+ARCHITECTURES="!x86_gcc2 x86_64 ?arm"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
 
 PROVIDES="
-	grep = $portVersion compat >= 3
-	cmd:egrep = $portVersion compat >= 2
-	cmd:fgrep = $portVersion compat >= 2
-	cmd:grep = $portVersion compat >= 2
+	grep$secondaryArchSuffix = $portVersion compat >= 3
+	cmd:egrep$commandSuffix = $portVersion compat >= 2
+	cmd:fgrep$commandSuffix = $portVersion compat >= 2
+	cmd:grep$commandSuffix = $portVersion compat >= 2
 	"
 REQUIRES="
-	haiku
-	lib:libiconv
-	lib:libintl
+	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libpcre$secondaryArchSuffix
 	"
```

Comment:
REPLACES is missing for x86_gcc2

Reply:
Need to see how to do that part right :) disabled x86_gcc2 in the previous versions, shouldn't that take care of it?

Reply:
No, grep_x86 should replace grep when it's already installed on x86_gcc2

Reply:
I think the below section should cover that now (at least it worked here locally)
Should I revert the changes to the prior version (eg enable x86_gcc2 again)?

Reply:
See the tar recipe for example:
https://github.com/haikuports/haikuports/commit/4b890dcedfcb01de02a18ec8ffa257595ade56ad#diff-14128caa66153dbe9da8ad6b8d6d46569e369c273edc7075a2a483a30e17a1a8R38

Reply:
Didn't think of that ;)